libdnf/dnf-context.cpp

```diff
@@ -1825,9 +1825,27 @@ dnf_context_set_os_release(DnfContext *context, GError **error) try
                                                  "os-release",
                                                  "VERSION_ID",
                                                  error);
-    if (maybe_quoted_version == NULL)
-        return FALSE;
-    version = g_shell_unquote(maybe_quoted_version, error);
+    if (maybe_quoted_version == NULL) {
+        /* rolling releases like Arch or Debian Unstable have no VERSION_ID */
+        g_autofree gchar *maybe_quoted_id = NULL;
+        maybe_quoted_id = g_key_file_get_string(key_file,
+                                                "os-release",
+                                                "ID",
+                                                error);
+        if (!maybe_quoted_id)
+            return FALSE;
+
+        g_autofree gchar *id = g_shell_unquote(maybe_quoted_id, error);
+        if (g_ascii_strncasecmp(id, "debian", strlen("debian")) != 0 &&
+                g_ascii_strncasecmp(id, "arch", strlen("arch")) != 0)
+            return FALSE;
+
+        /* Fake a high version number */
+        version = g_strdup("9999");
+        g_clear_error(error);
```

Comment:
If we get here then error should be NULL (right?), in which case we don't need this line.  Though it's probably a no-op anyway.